### ISSUE TITLE
Proposal: do not exit instantly on error

### ISSUE THREAD

**Issue by austenadler** [earthly/earthly#607]
Whenever there is a build failure in earthly, the process terminates immediately. It has happened before that a long apt upgrade is interrupted by an unrelated layer build. It would be nice to have an option to complete either as much as possible, or at least all currently running layers before exiting.

Reasoning:

- Speed/efficiency: When debugging a build, long running layers can complete and don't have to rerun each time. Also, if the layer was a download (apt-get update or apt-get upgrade download portion), the server won't get hit multiple times per retry
- Independence: One of the reasons earthly is so parallelizable is because layers are independent. So why should one unrelated layer kill another?

Proposal:
Option 1:

- Add an option like `--graceful-fail`. This will cause earthly not to queue new layer builds, but allow existing layers to complete before failing
- Will not repeat network requests or terminate long-running commands.

Option 2:

- Add an option like `--keep-going` (like from Gentoo's [emerge](https://dev.gentoo.org/~zmedico/portage/doc/man/emerge.1.html)). This will complete all layers that do not depend on the failed layer, then fails at the very end
- Allows for a "run and forget" strategy where the dev can come back, fix the broken layer `L`, and only `L` and dependent layers need to be built again

**Comment by andyli** [comment]
The current earthly behavour is problematic for terraform.

I'm trying to use multiple `BUILD` to `terraform apply` on two independent terraform modules. When one of the apply fail, 
 Earthly will kill the other one, preventing terraform to release its [state lock](https://www.terraform.io/docs/language/state/locking.html). I have to `terraform force-unlock ...` manually each time it happens.

**Comment by jdewar** [comment]
I was looking for this; I have layers defined like
```
A -> B
 \ C -> D
```
B fails often during code changes.  C only changes when dependencies change, but it takes a long time to build.  If I don't have C cached, I want C to continue building through to the end _even if B fails_.  That way subsequent code changes will start with B and D running in parallel -- the layers that actually test code edits.

Right now I do it manually.  I run the slow layer after I realize that it was still building when a `+all` run fails.

One solution would be to just make it A -> C -> B | D, but C'll take 10 min, and I'd rather be already ~8 minutes into fixing the bug that B alerted me to when C completes.

**Comment by vladaionescu** [comment]
Buildkit needs support for something like this before we can add this to Earthly, unfortunately. This would be useful though.

**Comment by glehmann** [comment]
in addition to terraform, this is also a problem with helm which require a manual intervention in that case. Clearly not ideal!
